cli/src/main.rs

```diff
@@ -29,9 +29,20 @@ pub enum DIDKit {
         #[structopt(flatten)]
         key: KeyArg,
     },
-    /// Output a DID for a given JWK and DID method name or pattern.
+    /// Output a DID for a given JWK according to the provided DID method name or pattern
+    ///
+    /// Deterministically generate a DID from a public key JWK, for a DID method
+    /// that support this functionality.
+    ///
+    /// The DID method to use may be provided in the `method-pattern` argument. The default is
+    /// `did:key`.
+    ///
+    /// For DID methods that have multiple ways of representing a key, `method-pattern` is
+    /// method-specific but typically is a prefix, for example "pkh:tz" to generate a DID that
+    /// begins with `did:pkh:tz`.
     KeyToDID {
-        /// DID method id or pattern. e.g. `key`, `tz`, or `pkh:tz`
+        /// DID method name or pattern. e.g. `key`, `tz`, or `pkh:tz`
+        #[structopt(default_value = "key")]
         method_pattern: String,
```

Comment:
Could that be an enum, just to show every possible choice from all resolvers enabled?

Reply:
That could be good.

We would have to keep the enum in sync with `DID_METHODS`. Another way could be to generate the list from `DID_METHODS`; e.g. adding a "enumerate_method_patterns" function to the DIDResolver trait. Some DID methods (i.e. `pkh`) accept multiple pattern strings.

I wonder if all of the possible method-patterns should appear in this help text, or if that should be reachable via another command or help text for another command.

Reply:
> We would have to keep the enum in sync with DID_METHODS. Another way could be to generate the list from DID_METHODS; e.g. adding a "enumerate_method_patterns" function to the DIDResolver trait. Some DID methods (i.e. pkh) accept multiple pattern strings.

It could be part of the resolver trait, and DID_METHODS is just the aggregation?

> I wonder if all of the possible method-patterns should appear in this help text, or if that should be reachable via another command or help text for another command.

Yeah I don't know, for now it's probably fine if it's in `--help` (maybe not `-h`). It reminds me a bit of `vim --version` which lists all compilation flags.

Reply:
> It could be part of the resolver trait, and DID_METHODS is just the aggregation?

Right.

> for now it's probably fine if it's in --help 

For this command, and for `key-to-verification-method`? Or somewhere else?

> (maybe not -h)

+1. I haven't yet been able to successfully define them separately though.

> It reminds me a bit of vim `--version` which lists all compilation flags.

+1, listing things could be useful. Supported DID methods, supported proof types, cargo features, etc.

Reply:
> For this command, and for key-to-verification-method? Or somewhere else?

Probably everywhere? Having an enum would be nice but not sure if there's a way to compose one (flattened) at compile time.

Reply:
new issue opened upon discussion